code/modules/vending/sustenance.dm

```diff
@@ -26,6 +26,9 @@
 	extra_price = PAYCHECK_LOWER * 0.6
 	payment_department = NO_FREEBIES
 
+	//Whether only prisoners can access the machine
+	var/prisoner_only = TRUE
```

Comment:
This isn't the right fix. The ID card has an `access` variable that contains all access levels. If it has either has centcom or prisoner access then allow vending.

You can do this by setting req_access of the machine to prisoner and centcom access and then use the proc allowed() instead of type checking for an ID card.

@MrMelbert that's the solution you want right?

Reply:
I don't think there is a prisoner access